### ISSUE TITLE
Standard Naming of DvObject types 

### ISSUE THREAD

**Issue by raprasad** [IQSS/dataverse#2490]
The project has 4 different naming conventions for the same object types.  Even if the number of naming conventions is not reduced, static variables should be used across the project.  (It would help to cut it to 2 or at least 3 naming conventions)

It is easy to make programming errors when comparing objects types across postgres, solr results, etc.

| Postgres dtype (db value and DvObject.java) | Solr Index Names (SearchConstants.java) | Search API   (SearchConstants.java) | Web UI as facets  (SearchConstants.java) |
| --- | --- | --- | --- |
| Dataverse | dataverses | dataverse | Dataverses |
| Dataset | datasets | dataset | Datasets |
| DataFile | files | file | Files |

Note: In addition, the static variables for the strings above are not always used.  Ther are many instances of simply using strings.

This is a side effect of the 'rapid development' in the winter
### Some examples of using strings

Dataverses.java

``` java
         jab.add( Json.createObjectBuilder().add("type", "dataverse")
```

Index.java

``` java
            if (type.equals("dataverses")) {
```

DataTagsAPI.java

``` java
        JsonObject jsonInfo = endpoint.queryParam("repositoryName", "Dataverse")
```

IndexServiceBean.java

``` java
       solrInputDocument.addField(SearchFields.TYPE, "dataverses");
```

SearchServiceBean.java

``` java
            if (type.equals("dataverses")) {
```

(etc, etc, etc)

**Comment by bencomp** [comment]
:+1: !

This is one for the (now closed) #775. I added it to the [Style Guide](https://docs.google.com/document/d/1KTd3FpM1BI3HlBofaZjMmBiQEJtFf11jiiGpQeJzy7A/edit):
> Use constants when comparing a string value with reference values

**Comment by pdurbin** [comment]
Yes, it's ugly but we're living with it. If someone wants to champion this issue, please open a new one. Closing.
